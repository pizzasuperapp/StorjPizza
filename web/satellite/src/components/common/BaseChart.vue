// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<script lang="ts">
import { Component, Prop, Vue, Watch } from 'vue-property-decorator';

// @vue/component
@Component
export default class BaseChart extends Vue {
    @Prop({default: 0})
    public readonly width: number;
    @Prop({default: 0})
    public readonly height: number;

    /**
     * Used for chart re rendering.
     */
    public chartKey = 0;

    /**
     * Triggers chart re rendering if chart width is being changed.
     */
    @Watch('width')
    public rebuildChart(): void {
        this.chartKey += 1;
    }
}
</script>
