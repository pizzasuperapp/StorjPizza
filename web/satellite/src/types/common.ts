// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
}

export enum OnboardingOS {
    WINDOWS = "windows",
    MAC = "macos",
    LINUX = "linux",
}

export class PartneredSatellite {
    constructor(
        public name: string = '',
        public address: string = '',
    ) {}
}
