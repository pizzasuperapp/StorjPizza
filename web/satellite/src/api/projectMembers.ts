// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { BaseGql } from '@/api/baseGql';
import { ProjectMember, ProjectMemberCursor, ProjectMembersApi, ProjectMembersPage } from '@/types/projectMembers';

export class ProjectMembersApiGql extends BaseGql implements ProjectMembersApi {

    /**
     * Used for adding team members to project.
     *
     * @param projectId
     * @param emails
     */
    public async add(projectId: string, emails: string[]): Promise<void> {
        const query =
            `mutation($projectId: String!, $emails:[String!]!) {
                addProjectMembers(
                    projectID: $projectId,
                    email: $emails
                ) {id}
            }`;

        const variables = {
            projectId,
            emails,
        };

        await this.mutate(query, variables);
    }

    /**
     * Used for deleting team members from project.
     *
     * @param projectId
     * @param emails
     */
    public async delete(projectId: string, emails: string[]): Promise<void> {
        const query =
            `mutation($projectId: String!, $emails:[String!]!) {
                deleteProjectMembers(
                    projectID: $projectId,
                    email: $emails
                ) {id}
            }`;

        const variables = {
            projectId,
            emails,
        };

        await this.mutate(query, variables);
    }

    /**
     * Used for fetching team members related to project.
     *
     * @param projectId
     * @param cursor for pagination
     */
    public async get(projectId: string, cursor: ProjectMemberCursor): Promise<ProjectMembersPage> {
        const query =
            `query($projectId: String!, $limit: Int!, $search: String!, $page: Int!, $order: Int!, $orderDirection: Int!) {
                project (
                    id: $projectId,
                ) {
                    members (
                        cursor: {
                            limit: $limit,
                            search: $search,
                            page: $page,
                            order: $order,
                            orderDirection: $orderDirection
                        }
                    ) {
                        projectMembers {
                            user {
                                id,
                                fullName,
                                shortName,
                                email
                            },
                            joinedAt
                        },
                        search,
                        limit,
                        order,
                        pageCount,
                        currentPage,
                        totalCount
                    }
                }
            }`;

        const variables = {
            projectId: projectId,
            limit: cursor.limit,
            search: cursor.search,
            page: cursor.page,
            order: cursor.order,
            orderDirection: cursor.orderDirection,
        };

        const response = await this.query(query, variables);

        return this.getProjectMembersList(response.data.project.members);
    }

    /**
     * Method for mapping project members page from json to ProjectMembersPage type.
     *
     * @param projectMembers anonymous object from json
     */
    private getProjectMembersList(projectMembers: any): ProjectMembersPage { // eslint-disable-line @typescript-eslint/no-explicit-any
        if (!projectMembers) {
            return new ProjectMembersPage();
        }

        const projectMembersPage: ProjectMembersPage = new ProjectMembersPage();
        projectMembersPage.projectMembers = projectMembers.projectMembers.map(key => new ProjectMember(
            key.user.fullName,
            key.user.shortName,
            key.user.email,
            new Date(key.joinedAt),
            key.user.id,
        ));

        projectMembersPage.search = projectMembers.search;
        projectMembersPage.limit = projectMembers.limit;
        projectMembersPage.order = projectMembers.order;
        projectMembersPage.orderDirection = projectMembers.orderDirection;
        projectMembersPage.pageCount = projectMembers.pageCount;
        projectMembersPage.currentPage = projectMembers.currentPage;
        projectMembersPage.totalCount = projectMembers.totalCount;

        return projectMembersPage;
    }
}
