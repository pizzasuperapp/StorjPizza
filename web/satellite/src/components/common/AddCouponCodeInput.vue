// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="add-coupon__input-container">
        <div v-if="!showConfirmMessage">
            <div class="add-coupon__input-wrapper">
                <img
                    class="add-coupon__input-icon"
                    :class="{'signup-view': isSignupView}"
                    src="@/../static/images/account/billing/coupon.png"
                    alt="Coupon"
                >
                <VInput
                    :label="inputLabel"
                    placeholder="Enter Coupon Code"
                    height="52px"
                    :with-icon="true"
                    @setData="setCouponCode"
                />
                <CheckIcon
                    v-if="isCodeValid"
                    class="add-coupon__check"
                />
            </div>
            <ValidationMessage
                class="add-coupon__valid-message"
                success-message="Successfully applied coupon code."
                :error-message="errorMessage"
                :is-valid="isCodeValid"
                :show-message="showValidationMessage"
            />
            <VButton
                v-if="!isSignupView"
                class="add-coupon__apply-button"
                label="Apply Coupon Code"
                width="85%"
                height="44px"
                :on-press="couponCheck"
            />
        </div>
        <div v-if="showConfirmMessage">
            <p class="add-coupon__confirm-message">
                By applying this coupon you will override your existing coupon.
                Are you sure you want to remove your current coupon and replace it with this new coupon?
            </p>
            <div class="add-coupon__button-wrapper">
                <VButton
                    class="add-coupon__confirm-button"
                    label="Yes"
                    width="250px"
                    height="44px"
                    :on-press="applyCouponCode"
                />
                <VButton
                    class="add-coupon__back-button"
                    label="Back"
                    width="250px"
                    height="44px"
                    is-blue-white="true"
                    :on-press="toggleConfirmMessage"
                />
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import VInput from '@/components/common/VInput.vue';
import ValidationMessage from '@/components/common/ValidationMessage.vue';
import VButton from '@/components/common/VButton.vue';

import CheckIcon from '@/../static/images/common/validCheck.svg';

import { PaymentsHttpApi } from '@/api/payments';
import { RouteConfig } from '@/router';
import { PAYMENTS_ACTIONS } from '@/store/modules/payments';

// @vue/component
@Component({
    components: {
        VButton,
        VInput,
        CheckIcon,
        ValidationMessage,
    },
})
export default class AddCouponCodeInput extends Vue {
    private showValidationMessage = false;
    private errorMessage = '';
    private isCodeValid = false;

    private couponCode = '';

    private showConfirmMessage = false;

    private readonly payments: PaymentsHttpApi = new PaymentsHttpApi();

    /**
     * Signup view requires some unque styling and element text.
     */
    public get isSignupView(): boolean {
        return this.$route.name === RouteConfig.Register.name;
    }

    /**
     * Returns label for input if in signup view
     * Depending on view.
     */
    public get inputLabel(): string | void {
        return this.isSignupView ? 'Add Coupon' : '';
    }

    public setCouponCode(value: string): void {
        this.couponCode = value;
    }

    /**
    * Toggles showing of coupon code replace confirmation message
    */
    public toggleConfirmMessage(): void {
        this.showConfirmMessage = !this.showConfirmMessage;
    }

    /**
     * Check if coupon code is valid
     */
    public async applyCouponCode(): Promise<void> {
        try {
            await this.$store.dispatch(PAYMENTS_ACTIONS.APPLY_COUPON_CODE, this.couponCode);
        }
        catch (error) {
            if (this.showConfirmMessage) {
                this.toggleConfirmMessage();
            }
            this.errorMessage = error.message;
            this.isCodeValid = false;
            this.showValidationMessage = true;

            return;
        }
        if (this.showConfirmMessage) {
            this.toggleConfirmMessage();
        }
        this.isCodeValid = true;
        this.showValidationMessage = true;
    }

    /**
     * Check if user has a coupon code applied to their account before applying
     */
    public async couponCheck(): Promise<void> {
        try {
            if (this.$store.state.paymentsModule.coupon) {
                this.toggleConfirmMessage();
            } else {
                this.applyCouponCode();
            }
        } catch (error) {

            this.errorMessage = error.message;
            this.isCodeValid = false;
            this.showValidationMessage = true;

            return;
        }
    }
}
</script>

<style scoped lang="scss">
    .add-coupon {

        &__input-wrapper {
            position: relative;
        }

        &__valid-message {
            position: relative;
            top: 15px;
        }

        &__apply-button {
            position: absolute;
            left: 0;
            right: 0;
            margin: 0 auto;
            bottom: 40px;
            background: #2683ff;

            &:hover {
                background: #0059d0;
            }
        }

        &__check {
            position: absolute;
            right: 15px;
            bottom: 15px;
        }

        &__input-icon {
            position: absolute;
            top: 27px;
            z-index: 23;
            left: 25px;
            width: 25.01px;
            height: 16.67px;
        }

        &__input-icon.signup-view {
            top: 63px;
        }

        &__confirm-message {
            font-family: 'font_regular', sans-serif;
            font-size: 18px;
            margin-top: 35px;
        }

        &__button-wrapper {
            display: flex;
            justify-content: space-evenly;
            margin-top: 30px;
        }
    }

</style>
