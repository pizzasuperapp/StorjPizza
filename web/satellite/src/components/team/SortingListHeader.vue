// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="sort-header-container">
        <div class="sort-header-container__name-container" @click="onHeaderItemClick(ProjectMemberOrderBy.NAME)">
            <p class="sort-header-container__name-container__title">Name</p>
            <VerticalArrows
                :is-active="areProjectMembersSortedByName"
                :direction="getSortDirection"
            />
        </div>
        <div class="sort-header-container__added-container" @click="onHeaderItemClick(ProjectMemberOrderBy.CREATED_AT)">
            <p class="sort-header-container__added-container__title">Added</p>
            <VerticalArrows
                :is-active="areProjectMembersSortedByDate"
                :direction="getSortDirection"
            />
        </div>
        <div class="sort-header-container__email-container" @click="onHeaderItemClick(ProjectMemberOrderBy.EMAIL)">
            <p class="sort-header-container__email-container__title">Email</p>
            <VerticalArrows
                :is-active="areProjectMembersSortedByEmail"
                :direction="getSortDirection"
            />
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import VerticalArrows from '@/components/common/VerticalArrows.vue';

import { SortDirection } from '@/types/common';
import { OnHeaderClickCallback, ProjectMemberOrderBy } from '@/types/projectMembers';

// @vue/component
@Component({
    components: {
        VerticalArrows,
    },
})
export default class SortingListHeader extends Vue {
    @Prop({default: () => () => new Promise(() => false)})
    private readonly onHeaderClickCallback: OnHeaderClickCallback;

    public ProjectMemberOrderBy = ProjectMemberOrderBy;

    public sortBy: ProjectMemberOrderBy = ProjectMemberOrderBy.NAME;
    public sortDirection: SortDirection = SortDirection.ASCENDING;

    /**
     * Used for arrow styling.
     */
    public get getSortDirection(): SortDirection {
        return this.sortDirection === SortDirection.DESCENDING ? SortDirection.ASCENDING : SortDirection.DESCENDING;
    }

    public get areProjectMembersSortedByName(): boolean {
        return this.sortBy === ProjectMemberOrderBy.NAME;
    }

    public get areProjectMembersSortedByDate(): boolean {
        return this.sortBy === ProjectMemberOrderBy.CREATED_AT;
    }

    public get areProjectMembersSortedByEmail(): boolean {
        return this.sortBy === ProjectMemberOrderBy.EMAIL;
    }

    /**
     * Sets sorting kind if different from current.
     * If same, changes sort direction.
     * @param sortBy
     */
    public async onHeaderItemClick(sortBy: ProjectMemberOrderBy): Promise<void> {
        if (this.sortBy !== sortBy) {
            this.sortBy = sortBy;
            this.sortDirection = SortDirection.ASCENDING;

            await this.onHeaderClickCallback(this.sortBy, this.sortDirection);

            return;
        }

        if (this.sortDirection === SortDirection.DESCENDING) {
            this.sortDirection = SortDirection.ASCENDING;
        } else {
            this.sortDirection = SortDirection.DESCENDING;
        }

        await this.onHeaderClickCallback(this.sortBy, this.sortDirection);
    }
}
</script>

<style scoped lang="scss">
    .sort-header-container {
        display: flex;
        flex-direction: row;
        height: 40px;
        background-color: rgb(255 255 255 / 30%);
        margin-top: 31px;

        &__name-container,
        &__added-container,
        &__email-container {

            &__title {
                font-family: 'font_medium', sans-serif;
                font-size: 16px;
                line-height: 23px;
                color: #2a2a32;
                margin: 0;
            }
        }

        &__name-container {
            display: flex;
            width: calc(50% - 70px);
            cursor: pointer;
            text-align: left;
            margin-left: 70px;
            align-items: center;
            justify-content: flex-start;
        }

        &__added-container {
            width: 25%;
            cursor: pointer;
            text-align: left;
            margin-left: 30px;
            display: flex;
            align-items: center;
            justify-content: flex-start;
        }

        &__email-container {
            width: 25%;
            cursor: pointer;
            text-align: left;
            display: flex;
            align-items: center;
            justify-content: flex-start;
        }
    }
</style>
