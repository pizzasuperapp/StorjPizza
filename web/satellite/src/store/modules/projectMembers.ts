// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { SortDirection } from '@/types/common';
import {
    ProjectMember,
    ProjectMemberCursor,
    ProjectMemberOrderBy,
    ProjectMembersApi,
    ProjectMembersPage,
} from '@/types/projectMembers';
import {StoreModule} from "@/types/store";

export const PROJECT_MEMBER_MUTATIONS = {
    FETCH: 'fetchProjectMembers',
    TOGGLE_SELECTION: 'toggleSelection',
    CLEAR_SELECTION: 'clearSelection',
    CLEAR: 'clearProjectMembers',
    CHANGE_SORT_ORDER: 'changeProjectMembersSortOrder',
    CHANGE_SORT_ORDER_DIRECTION: 'changeProjectMembersSortOrderDirection',
    SET_SEARCH_QUERY: 'setProjectMembersSearchQuery',
    SET_PAGE: 'setProjectMembersPage',
};

const {
    FETCH,
    TOGGLE_SELECTION,
    CLEAR_SELECTION,
    CLEAR,
    CHANGE_SORT_ORDER,
    CHANGE_SORT_ORDER_DIRECTION,
    SET_SEARCH_QUERY,
    SET_PAGE,
} = PROJECT_MEMBER_MUTATIONS;

export class ProjectMembersState {
    public cursor: ProjectMemberCursor = new ProjectMemberCursor();
    public page: ProjectMembersPage = new ProjectMembersPage();
    public selectedProjectMembersEmails: string[] = [];
}

interface ProjectMembersContext {
    state: ProjectMembersState
    commit: (string, ...unknown) => void
    rootGetters: {
        selectedProject: {
            id: string
        }
    }
}

export function makeProjectMembersModule(api: ProjectMembersApi): StoreModule<ProjectMembersState, ProjectMembersContext> {
    return {
        state: new ProjectMembersState(),
        mutations: {
            [FETCH](state: ProjectMembersState, page: ProjectMembersPage) {
                state.page = page;
                state.page.projectMembers = state.page.projectMembers.map(member => {
                    if (state.selectedProjectMembersEmails.includes(member.user.email)) {
                        member.isSelected = true;
                    }

                    return member;
                });
            },
            [SET_PAGE](state: ProjectMembersState, page: number) {
                state.cursor.page = page;
            },
            [SET_SEARCH_QUERY](state: ProjectMembersState, search: string) {
                state.cursor.search = search;
            },
            [CHANGE_SORT_ORDER](state: ProjectMembersState, order: ProjectMemberOrderBy) {
                state.cursor.order = order;
            },
            [CHANGE_SORT_ORDER_DIRECTION](state: ProjectMembersState, direction: SortDirection) {
                state.cursor.orderDirection = direction;
            },
            [CLEAR](state: ProjectMembersState) {
                state.cursor = new ProjectMemberCursor();
                state.page = new ProjectMembersPage();
                state.selectedProjectMembersEmails = [];
            },
            [TOGGLE_SELECTION](state: ProjectMembersState, projectMember: ProjectMember) {
                if (!state.selectedProjectMembersEmails.includes(projectMember.user.email)) {
                    projectMember.isSelected = true;
                    state.selectedProjectMembersEmails.push(projectMember.user.email);

                    return;
                }

                projectMember.isSelected = false;
                state.selectedProjectMembersEmails = state.selectedProjectMembersEmails.filter(projectMemberEmail => {
                    return projectMemberEmail !== projectMember.user.email;
                });
            },
            [CLEAR_SELECTION](state: ProjectMembersState) {
                state.selectedProjectMembersEmails = [];
                state.page.projectMembers = state.page.projectMembers.map((projectMember: ProjectMember) => {
                    projectMember.isSelected = false;

                    return projectMember;
                });
            },
        },
        actions: {
            addProjectMembers: async function ({rootGetters}: ProjectMembersContext, emails: string[]): Promise<void> {
                const projectId = rootGetters.selectedProject.id;

                await api.add(projectId, emails);
            },
            deleteProjectMembers: async function ({rootGetters, state, commit}: ProjectMembersContext): Promise<void> {
                const projectId = rootGetters.selectedProject.id;

                await api.delete(projectId, state.selectedProjectMembersEmails);

                commit(CLEAR_SELECTION);
            },
            fetchProjectMembers: async function ({commit, rootGetters, state}: ProjectMembersContext, page: number): Promise<ProjectMembersPage> {
                const projectID = rootGetters.selectedProject.id;

                commit(SET_PAGE, page);

                const projectMembersPage: ProjectMembersPage = await api.get(projectID, state.cursor);

                commit(FETCH, projectMembersPage);

                return projectMembersPage;
            },
            setProjectMembersSearchQuery: function ({commit}: ProjectMembersContext, search: string) {
                commit(SET_SEARCH_QUERY, search);
            },
            setProjectMembersSortingBy: function ({commit}: ProjectMembersContext, order: ProjectMemberOrderBy) {
                commit(CHANGE_SORT_ORDER, order);
            },
            setProjectMembersSortingDirection: function ({commit}: ProjectMembersContext, direction: SortDirection) {
                commit(CHANGE_SORT_ORDER_DIRECTION, direction);
            },
            clearProjectMembers: function ({commit}: ProjectMembersContext) {
                commit(CLEAR);
                commit(CLEAR_SELECTION);
            },
            toggleProjectMemberSelection: function ({commit}: ProjectMembersContext, projectMember: ProjectMember) {
                commit(TOGGLE_SELECTION, projectMember);
            },
            clearProjectMemberSelection: function ({commit}: ProjectMembersContext) {
                commit(CLEAR_SELECTION);
            },
        },
        getters: {
            selectedProjectMembers: (state: ProjectMembersState) =>
                state.page.projectMembers.filter((member: ProjectMember) =>
                    state.selectedProjectMembersEmails.includes(member.user.email)),
        },
    };
}
