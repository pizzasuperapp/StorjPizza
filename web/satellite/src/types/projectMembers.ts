// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * ProjectMember stores needed info about user info to show it on UI.
 */
import { SortDirection } from '@/types/common';
import { User } from '@/types/users';

export type OnHeaderClickCallback = (sortBy: ProjectMemberOrderBy, sortDirection: SortDirection) => Promise<void>;

export enum ProjectMemberOrderBy {
    NAME = 1,
    EMAIL,
    CREATED_AT,
}

/**
 * Contains values of project members header component state
 * used in ProjectMembersArea and HeaderArea.
 */
export enum ProjectMemberHeaderState {
    DEFAULT = 0,
    /**
     * Used when some project members selected
     */
    ON_SELECT,
}

/**
 * ProjectMembersApi is a graphql implementation of ProjectMembers API.
 * Exposes all ProjectMembers-related functionality
 */
export interface ProjectMembersApi {
    /**
     * Add members to project by user emails.
     *
     * @param projectId
     * @param emails list of project members email to add
     *
     * @throws Error
     */
    add(projectId: string, emails: string[]): Promise<void>;

    /**
     * Deletes ProjectMembers from project by project member emails
     *
     * @param projectId
     * @param emails
     *
     * @throws Error
     */
    delete(projectId: string, emails: string[]): Promise<void>;

    /**
     * Fetch Project Members
     *
     * @param projectId
     * @param cursor
     *
     * @throws Error
     */
    get(projectId: string, cursor: ProjectMemberCursor): Promise<ProjectMembersPage>;
}

/**
 * ProjectMemberCursor is a type, used for paged project members request.
 */
export class ProjectMemberCursor {
    public constructor(
        public search: string = '',
        public limit: number = 6,
        public page: number = 1,
        public order: ProjectMemberOrderBy = ProjectMemberOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
    ) {}
}

/**
 * ProjectMembersPage is a type, used to describe paged project members list.
 */
export class ProjectMembersPage {
    public constructor(
        public projectMembers: ProjectMember[] = [],
        public search: string = '',
        public order: ProjectMemberOrderBy = ProjectMemberOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
        public limit: number = 6,
        public pageCount: number = 0,
        public currentPage: number = 1,
        public totalCount: number = 0,
    ) {}
}

/**
 * ProjectMember is a type, used to describe project member.
 */
export class ProjectMember {
    public user: User;
    public isSelected: boolean;

    public constructor(
        public fullName: string = '',
        public shortName: string = '',
        public email: string = '',
        public joinedAt: Date = new Date(),
        public id: string = '',
    ) {
        this.user = new User(this.id, this.fullName, this.shortName, this.email);
        this.isSelected = false;
    }

    /**
     * Returns user's full name.
     */
    public get name(): string {
        return this.user.getFullName();
    }

    /**
     * Returns joined at date as a local date string.
     */
    public localDate(): string {
        return this.joinedAt.toLocaleDateString();
    }
}
