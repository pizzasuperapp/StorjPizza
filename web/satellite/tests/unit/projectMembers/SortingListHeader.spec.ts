// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import SortingListHeader from '@/components/team/SortingListHeader.vue';

import { SortDirection } from '@/types/common';
import { ProjectMemberOrderBy } from '@/types/projectMembers';
import { mount } from '@vue/test-utils';

describe('SortingListHeader.vue', () => {
    it('should render correctly', function () {
        const wrapper = mount(SortingListHeader);

        expect(wrapper).toMatchSnapshot();
    });

    it('should retrieve callback', function () {
        const onPressSpy = jest.fn();

        const wrapper = mount(SortingListHeader, {
            propsData: {
                onHeaderClickCallback: onPressSpy,
            },
        });
        wrapper.find('.sort-header-container__name-container').trigger('click');
        expect(onPressSpy).toHaveBeenCalledTimes(1);
    });

    it('should change sort direction', function () {
        const onPressSpy = jest.fn();

        const wrapper = mount(SortingListHeader, {
            propsData: {
                onHeaderClickCallback: onPressSpy,
            },
        });

        expect(wrapper.vm.$data.sortBy).toBe(ProjectMemberOrderBy.NAME);
        expect(wrapper.vm.$data.sortDirection).toBe(SortDirection.ASCENDING);

        wrapper.find('.sort-header-container__name-container').trigger('click');
        expect(onPressSpy).toHaveBeenCalledTimes(1);

        expect(wrapper.vm.$data.sortBy).toBe(ProjectMemberOrderBy.NAME);
        expect(wrapper.vm.$data.sortDirection).toBe(SortDirection.DESCENDING);
    });

    it('should change sort by value', function () {
        const onPressSpy = jest.fn();

        const wrapper = mount(SortingListHeader, {
            propsData: {
                onHeaderClickCallback: onPressSpy,
            },
        });

        expect(wrapper.vm.$data.sortBy).toBe(ProjectMemberOrderBy.NAME);
        expect(wrapper.vm.$data.sortDirection).toBe(SortDirection.ASCENDING);

        wrapper.find('.sort-header-container__added-container').trigger('click');
        expect(onPressSpy).toHaveBeenCalledTimes(1);

        expect(wrapper.vm.$data.sortBy).toBe(ProjectMemberOrderBy.CREATED_AT);
        expect(wrapper.vm.$data.sortDirection).toBe(SortDirection.ASCENDING);
    });
});
