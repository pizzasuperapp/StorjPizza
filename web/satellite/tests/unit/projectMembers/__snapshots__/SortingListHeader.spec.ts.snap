// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingListHeader.vue should render correctly 1`] = `
<div class="sort-header-container">
  <div class="sort-header-container__name-container">
    <p class="sort-header-container__name-container__title">Name</p>
    <div class="container"><svg class=""></svg> <svg class="active"></svg></div>
  </div>
  <div class="sort-header-container__added-container">
    <p class="sort-header-container__added-container__title">Added</p>
    <div class="container"><svg class=""></svg> <svg class=""></svg></div>
  </div>
  <div class="sort-header-container__email-container">
    <p class="sort-header-container__email-container__title">Email</p>
    <div class="container"><svg class=""></svg> <svg class=""></svg></div>
  </div>
</div>
`;
